devicetypes/sensative/sensative-strips-drip-700.src/sensative-strips-drip-700.groovy

```diff
@@ -0,0 +1,559 @@
+	}
+
+	if (state.pendingRefresh || (state.wakeUpInterval != wakeUpIntervalSeconds)) {
+		logDebug "Changing wake up interval to ${wakeUpIntervalSeconds} seconds"
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalSet(seconds:wakeUpIntervalSeconds, nodeid:zwaveHubNodeId))
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalGet())
+	}
+
+	configParams.each {
+		Integer storedVal = getParamStoredValue(it.num)
+		Integer settingVal = getSettingValue(it.num)
+
+		if (it.num != leakageCalibrationParamNum) {
+			if ((settingVal != null) && (settingVal != storedVal)) {
+				logDebug "CHANGING ${it.name}(#${it.num}) from ${storedVal} to ${settingVal}"
+				cmds << configSetCmd(it, settingVal)
+				cmds << configGetCmd(it)
+			} else if (state.pendingRefresh) {
+				cmds << configGetCmd(it)
+			}
+		} else {
+			if (settingVal && !state.leakageCalibrated) {
+				logDebug "Performing leakage/moisture sensor calibration..."
+				state.leakageCalibrated = false // Indicate that calibration has been started
+				cmds << configSetCmd(it, settingVal)
+				cmds << configGetCmd(it)
+			}
+		}
+	}
+
+	state.pendingRefresh = false
+	return cmds
+}
+
+
+// Required for HealthCheck Capability, but doesn't actually do anything because this device sleeps.
+def ping() {
+	logDebug "ping()"
+}
+
+
+def refresh() {
+	logDebug "refresh()..."
+	state.pendingRefresh = true
+	logForceWakeupMessage("The device will be refreshed the next time it wakes up.")
+}
+
+void logForceWakeupMessage(String msg) {
+	log.warn "${msg}  To force the device to wake up immediately, move the magnet towards the round end 3 times."
+}
+
+
+String batteryGetCmd() {
+	return secureCmd(zwave.batteryV1.batteryGet())
+}
+
+String configSetCmd(Map param, int value) {
+	return secureCmd(zwave.configurationV2.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: value))
+}
+
+String configGetCmd(Map param) {
+	return secureCmd(zwave.configurationV2.configurationGet(parameterNumber: param.num))
+}
+
+String secureCmd(cmd) {
+	try {
+		if (zwaveInfo?.zw?.contains("s")) {
+			return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			return cmd.format()
+		}
+	} catch (ex) {
+		return cmd.format()
+	}
+}
+
+void sendCommands(List<String> cmds, Integer delay=250) {
+	if (cmds) {
+		def actions = []
+		cmds.each {
+			actions << new physicalgraph.device.HubAction(it)
+		}
+		sendHubCommand(actions, delay)
+	}
+}
+
+
+def parse(String description) {
+	def cmd = zwave.parse(description, commandClassVersions)
+	if (cmd) {
+		zwaveEvent(cmd)
+	} else {
+		log.warn "Unable to parse: $description"
+	}
+
+	updateLastCheckIn()
+	return []
+}
+
+void updateLastCheckIn() {
+	if (!isDuplicateCommand(state.lastCheckInTime, 60000)) {
+		state.lastCheckInTime = new Date().time
+
+		sendEvent(name: "lastCheckIn", value: convertToLocalTimeString(new Date()), displayed: false)
```

Comment:
what's the purpose of this attribute?

Reply:
@krlaframboise, do you remember the purpose for this functionality?

Reply:
@dparyani @krlaframboise Please, could You refer to @greens question ?

Reply:
A long time ago, probably before the Health Check capability was working properly, the device's last activity date wouldn't update unless an event was created.  Updating that attribute in the parse method ensured that the date stayed updated, but it's really not needed anymore... 

@dparyani, the following can be deleted:

**lines 284-303:**
void updateLastCheckIn() {
	if (!isDuplicateCommand(state.lastCheckInTime, 60000)) {
		state.lastCheckInTime = new Date().time

		sendEvent(name: "lastCheckIn", value: convertToLocalTimeString(new Date()), displayed: false)
	}
}

String convertToLocalTimeString(dt) {
	try {
		def timeZoneId = location?.timeZone?.ID
		if (timeZoneId) {
			return dt.format("MM/dd/yyyy hh:mm:ss a", TimeZone.getTimeZone(timeZoneId))
		} else {
			return "$dt"
		}
	} catch (ex) {
		return "$dt"
	}
}

**line 280:**
updateLastCheckIn()

**line 85:**
attribute "lastCheckIn", "string"

Reply:
@greens and @KKlimczukS We have made the changes as suggested by @krlaframboise and sanity tested the device handlers. Please continue with the review. Thank you!